### DISCUSSION TITLE
Use GraalVM's Truffle

### DISCUSSION THREAD

**Discussion by nightscape** [mmhelloworld/idris-jvm#114] [Ideas]
Hi @mmhelloworld,

I'm wondering if you have considered Truffle as an alternative to the current code generation approach.
The performance of the corresponding Ruby, Python and R implementations is amazing!

**Comment by mmhelloworld** [comment]
Hi @nightscape ,

I did look into it a bit a while ago but since I wasn't much familiar with Truffle, bytecode generation seemed to be the faster way to get Idris up and running on the JVM. It definitely looks interesting with all the optimizations it enables. I am interested in trying it out some time for Idris.
